README.md

```diff
@@ -1,16 +1,110 @@
-## exclusive-lock-node
+<!-- TOC -->
 
+- [`ExclusiveLock`](#exclusivelock)
+- [Installation](#installation)
+- [Usage](#usage)
+- [API](#api)
+  - [`new ExclusiveLock(options)`](#new-exclusivelockoptions)
+  - [`acquire()`](#acquire)
+  - [`inspect()`](#inspect)
+  - [`release()`](#release)
+  - [`refresh()`](#refresh)
+- [Authors](#authors)
+- [License](#license)
 
+<!-- /TOC -->
+## `ExclusiveLock`
+
+Need to synchronize a bunch of workers so that only 1 does a particular thing?  If yes,
+then you need a lock file.  This package uses a cache store (`redis` or `keydb`)
+to create records atomically that only 1 instance will "own."
 ## Installation
 
+```bash
+npm install @logdna/exclusive-lock --save
+```
 
 ## Usage
 
+```js
+const ExclusiveLock = require('@logdna/exclusive-lock')
+const cache_connection = require('./my-keydb-connection.js')
+const pino = require('pino')
+
+const exclusive_lock = new ExclusiveLock({
+  app_name: 'my-distributed-app'
+, log: pino()
+, cache_manager
+})
+
+async function main() {
+  const got_lock = await exclusive_lock.acquire()
+  if (got_lock) {
+    log.info('You win the race! Lock acquired')
+  }
+  exclusive_lock.release()
+}
+
+main()
+```
+
+## API
+
+-------
+
+### `new ExclusiveLock(options)`
+* `options` [`<Object>`][]
+  * `app_name` [`<String>`][] - A unique string to identify your application
```

Comment:
This feels like some internal carry over, the use of this may not be used in the context of an application.
I think we should just call this `name`. make it not required, and default it to some random string or a uid.

Reply:
Yeah, agreed, but this is the most important part of the code. Multiple instances have to share the same name so it cannot be generated.

We cannot call this `name` if lock_name will also become `name`.  Maybe this one is an identifier? That's not great either because it implies an alphanumeric arbitrary string as opposed to a proper name.

Maybe lock_name should be `key`?  Solvedit